File: program/program/find-sum-of-n-natural-numbers-by-recursion/FindSumOfNNaturalNumbersByRecursion.kt
fun main() {
    print("Input : ")
    println()
    var input = readLine()

    println("Sum of natural number $input using recursion is : " + FindSumOfNNaturalNumbersByRecursion(input!!.toInt()))

}

private fun FindSumOfNNaturalNumbersByRecursion(num: Int): Int {
    val sum = addNumbers(num)

    return sum
}

private fun addNumbers(num: Int): Int {
    if (num != 0)
        return num + addNumbers(num - 1)
    else
        return num
}

File: program/program/find-sum-of-n-natural-numbers-by-recursion/FindSumOfNNaturalNumbersByRecursion.js
function sumOfNNaturalNumber(num) {
    let sumOfNaturalNum = 0;
    if(num > 0) {
        sumOfNaturalNum = num + sumOfNNaturalNumber(num - 1);
    }
    return sumOfNaturalNum;
}

console.log(sumOfNNaturalNumber(7));
